src/common/tracer.cc

```diff
@@ -55,6 +65,50 @@ bool Tracer::is_enabled() const {
   return g_ceph_context->_conf->jaeger_tracing_enable;
 }
 
+void encode(const jspan_context& span_ctx, bufferlist& bl, uint64_t f) {
+  if (span_ctx.IsValid()) {
+    using namespace opentelemetry;
+    using namespace trace;
+    const size_t ctx_size = TraceId::kSize + SpanId::kSize + 1;
+    char trace_identity[ctx_size];
+    uint8_t* trace_identity_ptr = (uint8_t*)trace_identity;
+    nostd::span<uint8_t, TraceId::kSize> bdika{trace_identity_ptr, TraceId::kSize};
+    span_ctx.trace_id().CopyBytesTo(bdika);
+    trace_identity_ptr += TraceId::kSize;
+    span_ctx.span_id().CopyBytesTo(nostd::span<uint8_t, SpanId::kSize>{trace_identity_ptr, SpanId::kSize});
+    trace_identity_ptr += SpanId::kSize;
+    span_ctx.trace_flags().CopyBytesTo(nostd::span<uint8_t, 1>{trace_identity_ptr, 1});
+
+    ENCODE_START(1, 1, bl);
+    encode(std::string_view(trace_identity, ctx_size), bl);
+    ENCODE_FINISH(bl);
+  }
+}
+
+void decode(jspan_context& span_ctx, bufferlist::const_iterator& bl) {
+  if (bl.get_remaining() == 0) {
+    return ;
+  }
+  using namespace opentelemetry;
+  using namespace trace;
+
+  const auto ctx_size = TraceId::kSize + SpanId::kSize + 1;
+  std::string trace_identity_string;
+  trace_identity_string.reserve(ctx_size);
+
+  DECODE_START(1, bl);
+  decode(trace_identity_string, bl);
+  DECODE_FINISH(bl);
+
+  uint8_t* trace_identity_ptr = (uint8_t*)trace_identity_string.data();
+  if (trace_identity_ptr) {
+    nostd::span<uint8_t, TraceId::kSize> trace_id(trace_identity_ptr, TraceId::kSize);
+    nostd::span<uint8_t, SpanId::kSize> span_id{trace_identity_ptr + TraceId::kSize, SpanId::kSize};
+    uint8_t flags = *(trace_identity_ptr + TraceId::kSize + SpanId::kSize);
+    SpanContext ctx(TraceId(trace_id), SpanId(span_id), TraceFlags(flags), true);
```

Comment:
why do you need the temporary span context? can you assign the values directly to `span_ctx`?

Reply:
no. those are private members of and there are no setters.

Reply:
then probably use: 
```cpp
span_ctx = std::move(ctx);
```